huaweicloud/services/ges/resource_huaweicloud_ges_metadata.go

```diff
@@ -0,0 +1,493 @@
+	}
+
+	getMetadataListPath := getMetadataListClient.Endpoint + getMetadataListHttpUrl
+	getMetadataListPath = strings.ReplaceAll(getMetadataListPath, "{project_id}", getMetadataListClient.ProjectID)
+
+	getMetadataListPath += "?limit=100" // user only can add 50 metadatas.
+
+	getMetadataListOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json;charset=UTF-8"},
+	}
+
+	getMetadataListResp, err := getMetadataListClient.Request("GET", getMetadataListPath, &getMetadataListOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getMetadataListRespBody, err := utils.FlattenResponse(getMetadataListResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jsonPath := fmt.Sprintf("schema_list[?id=='%s']|[0]", d.Id())
+	getMetadataListRespBody = utils.PathSearch(jsonPath, getMetadataListRespBody, nil)
+	if getMetadataListRespBody == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "no data found")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", getMetadataListRespBody, nil)),
+		d.Set("metadata_path", utils.PathSearch("metadata_path", getMetadataListRespBody, nil)),
+		d.Set("description", utils.PathSearch("description", getMetadataListRespBody, nil)),
+		d.Set("status", utils.PathSearch("status", getMetadataListRespBody, nil)),
+		d.Set("encryption", flattenGetMetadatasRespBodyEncryption(getMetadataListRespBody)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenGetMetadatasRespBodyEncryption(resp interface{}) []interface{} {
+	rst := []interface{}{
+		map[string]interface{}{
+			"enable":        utils.PathSearch("encrypted", resp, nil),
+			"master_key_id": utils.PathSearch("master_key_id", resp, nil),
+		},
+	}
+	return rst
+}
+
+func flattenGetMetadataRespBodyMetadata(resp interface{}) []interface{} {
+	var rst []interface{}
+	curJson, err := jmespath.Search("gesMetadata", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing gesMetadata from response= %#v", resp)
+		return rst
+	}
+
+	rst = []interface{}{
+		map[string]interface{}{
+			"labels": flattenMetadataLabels(curJson),
+		},
+	}
+	return rst
+}
+
+func flattenMetadataLabels(resp interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("labels", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		rst = append(rst, map[string]interface{}{
+			"name":       utils.PathSearch("name", v, nil),
+			"properties": utils.PathSearch("properties", v, nil),
+		})
+	}
+	return rst
+}
+
+func resourceGesMetadataUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateMetadataChanges := []string{
+		"name",
```

Comment:
looks name should be forcenew and used to distinguish the metadata when updating